test/onnx/test_fx_dynamic_with_onnxruntime.py

```diff
@@ -0,0 +1,384 @@
+        # model cannot be called with kwargs.
+        bound = signature.bind(*model_input_args, **model_input_kwargs)
+        # Fill optional inputs.
+        bound.apply_defaults()
+        assert not bound.kwargs
+
+        pt_cloned_model = _try_clone_model(model)
+        ref_outputs, _ = pytree.tree_flatten(
+            pt_cloned_model(*model_input_args, **model_input_kwargs)
+        )
+        ort_outputs = _run_ort(onnx_model, bound.args)
+        for ref_output, ort_output in zip(ref_outputs, ort_outputs):
+            torch.testing.assert_close(
+                ref_output, torch.tensor(ort_output), rtol=rtol, atol=atol
+            )
+
+    # Feed args and kwargs into exporter.
+    # Note that exporter should flatten kwargs into positional args the exported model;
+    # since ONNX doesn't represent kwargs.
+    onnx_model = fx_onnx.export_after_normalizing_args_and_kwargs(
+        model,
+        *input_args,
+        opset_version=opset_version,
+        use_binary_format=True,
+        dynamic_axes=True,  # export models with dynamic shapes
+        **input_kwargs,
+    )
+
+    compare_pytorch_onnx_with_ort(onnx_model, input_args, input_kwargs)
+
+    # TODO(titaiwang): do not support kwargs now
+    if additional_test_inputs:
+        for additional_input_args in additional_test_inputs:
+            compare_pytorch_onnx_with_ort(onnx_model, additional_input_args, {})
+
+
+class TestFxDynamicWithOnnxRuntime(onnx_test_common._TestONNXRuntime):
+    def setUp(self):
+        super().setUp()
+        self.diag_ctx = diagnostics.engine.create_diagnostic_context(
+            "test_fx_export", version=torch.__version__
+        )
+        self.opset_version = 18
+
+    def tearDown(self):
+        diagnostics.engine.dump(
+            f"test_report_{self._testMethodName}.sarif", compress=False
+        )
+        super().tearDown()
+
+    @unittest.skip(
+        "_aten_convolution_onnx: _add_attribute_to_torchscript_node()"
+        " parameter value=[None, None] violates type hint"
+        "typing.Union[float, int, str, bytes, typing.Sequence[float],"
+        " typing.Sequence[int], torch.Tensor], as [None, None]:"
+    )
+    def test_shufflenet_v2_dynamic_axes(self):
+        model = torchvision.models.shufflenet_v2_x0_5(pretrained=False)
+        dummy_input = torch.randn(1, 3, 224, 224, requires_grad=True)
+        test_inputs = torch.randn(3, 3, 224, 224, requires_grad=True)
+
+        _run_test_with_fx_to_onnx_exporter_and_onnx_runtime(
+            model,
+            (dummy_input,),
+            additional_test_inputs=[(dummy_input,), (test_inputs,)],
+            rtol=1e-3,
+            atol=1e-5,
+        )
+
+    def test_add(self):
+        class DynamicAdd(torch.nn.Module):
+            def forward(self, x, y):
+                return torch.ops.aten.add(x, y)
+
+        x = torch.randn(2, 3)
+        y = torch.randn(2, 3)
+        input_x = torch.randn(3, 4)
```

Comment:
```suggestion
        another_x = torch.randn(3, 4)
```

Reply:
It's kind of confusing seeing two input `x`..

Reply:
Done